dmscripts/json_schema_service_faker.py

```diff
@@ -0,0 +1,183 @@
+        return None
+
+    def __boolean(self, schema):
+        return random.choice([True, False])
+
+    def __enum(self, schema):
+        return random.choice(schema['enum'])
+
+    def __array(self, schema):
+        """Handles JSON Schema 'array' type: essentially a Python list"""
+        min_items = schema.get('minItems', 1)
+        max_items = schema.get('maxItems', 10)
+        num_items = random.randint(min_items, max_items)
+
+        if schema.get('uniqueItems', False):
+            items = set()
+            add_to_items = items.add
+        else:
+            items = []
+            add_to_items = items.append
+
+        attempts = 0
+        while len(items) < num_items:
+            len_items = len(items)
+
+            add_to_items(self.__parse_definition(schema['items']))
+
+            # Prevent infinite loops
+            if len_items == len(items):
+                attempts += 1
+                if attempts >= self.__max_recalcs:
+                    break
+            else:
+                attempts = 0
+
+        return list(items)
+
+    def __string(self, schema):
+        """Handles JSON Schema 'string' type: essentially a Python str"""
+        min_length = schema.get('minLength', 0)
+        max_length = schema.get('maxLength', 100)
+
+        random_string = lorem.text()[:random.randint(min_length, max_length)]
+
+        if schema.get('pattern'):
+            random_string = self.__map_to_function('pattern')(random_string,
+                                                              schema['pattern'],
+                                                              min_length=min_length,
+                                                              max_length=max_length)
+
+        elif schema.get('format'):
+            random_string = self.__map_to_function(schema['format'])(min_length=min_length,
+                                                                     max_length=max_length)
+
+        return random_string
+
+    def __pattern(self, current_string, pattern, min_length, max_length):
+        """Generates a random string matching a given regex pattern, with min/max length constraints."""
+        if re.match(pattern, current_string):
+            return current_string
+
+        random_string = ''
+        attempts = 0
+
+        while attempts < self.__max_recalcs:
+            random_string = xeger.xeger(pattern)
+
+            if min_length < len(random_string) < max_length:
+                return random_string
+
+            attempts += 1
+
+        logger.warning("Unable to generate within length constraints ({} < len < {}) for pattern: '{}'"
+                       .format(min_length, max_length, pattern))
+
+        return random_string
+
+    def __uri(self, min_length, max_length):
+        """Just returns a single URL; g-9 services' compulsory fields only use URI for uploaded documents."""
+        return 'https://www.digitalmarketplace.service.gov.uk/uploaded_document.pdf'
```

Comment:
It does... But also don't really expect the class to ever get re-used, so it's a good way to prompt a proper review before trying to jerry-rig it to do something else.

But I'm just trying to justify my bad decision here...